### PR TITLE
WIP: 1D structure function support

### PR DESCRIPTION
**Pull request by cassidymwagner**
This functionality will allow the user to calculate structure functions for 1D data. Advective structure functions are not supported. 

Current status:
- [x] SFs for 1D simulated and evenly gridded data
- [x] Uneven grid support (i.e. lat-lon)
- [x] Tests
- [x] Documentation
    - [x] Docstring in progress, read-through before merging
    - [x] Jupyter notebook example

### CODE REVIEW

**Comment by BrodiePearson** [comment]
@cassidymwagner This looks good to me so far!

File: fluidsf/generate_structure_functions_1d.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+from .bin_data import bin_data
+from .calculate_separation_distances import calculate_separation_distances
+from .calculate_structure_function_1d import calculate_structure_function_1d
+from .shift_array1d import shift_array1d
+
+
+def generate_structure_functions_1d(  # noqa: C901, D417
+    u,
+    v,
```
Comment (cassidymwagner) [inline fluidsf/generate_structure_functions_1d.py:11]: Technically `v` is only necessary for `LTT` structure functions, so it should probably be optional.

Comment (cassidymwagner) [reply fluidsf/generate_structure_functions_1d.py]: Added to 445539a.

**Comment by cassidymwagner** [comment]
I'm going to merge this since the tests all pass and the modules should be consistent with the 2D versions in main now. There's still a little work to do with documentation but I'll continue that work on #23.

**Pull request merged**